File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    answer = []
    ccnt = 0
    zcnt = 0
    while True:
        tmp = ''
        ccnt += 1
        for i in s:
            if i == '1':
                tmp += i
            else:
                zcnt += 1
        s = bin(len(tmp))[2:]
        if s == '1':
            break
    return [ccnt,zcnt]

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(string s) {
    vector<int> answer(2,0);
    int ccnt = 0, zcnt = 0;
    
    while (s != "1"){
        ccnt++;
        string tmp = "";
        int num;
        for (char c: s){
            if (c == '0'){
                zcnt++;
                }
            else{
                tmp += c;
            }
        } 
        num = tmp.size();
        s = "";
        while(num > 0){
            s += to_string(num%2);
            num /= 2; 
        }
    }
    answer[0] = ccnt;
    answer[1] = zcnt;
    return answer;
}